src/extendDescriptor.js

```diff
@@ -1,18 +1,83 @@
 import { decorate } from './private/utils';
 
 function handleDescriptor(target, key, descriptor) {
-  const superKlass = Object.getPrototypeOf(target);
-  const superDesc = Object.getOwnPropertyDescriptor(superKlass, key);
+  if (typeof descriptor.value === 'function') {
+    return handleMethod(target, key, descriptor);
+  }
+  else if (typeof descriptor.initializer === 'function') {
+    throw new Error(`unable to decorate initialized property "${key}"`); 
+    //return handleInitializedProperty(target, key, descriptor);
+  }
+  else {
+    return handleProperty(target, key, descriptor);
+  }
+}
+
+/*
+function handleInitializedProperty(target, key, descriptor{
+  if (typeof target === 'function') {
+  }
+  else
+  if (typeof target === 'object') {
+  }
+
+  return {
+    ...superDesc,
+    initializer: descriptor.initializer
+  };
+}
+*/
+
+function handleMethod(target, key, descriptor) {
+  const superDesc = findSuperDesc(target, key);
+
+  return {
+    ...superDesc,
+    value: descriptor.value
+  };
+}
+
+function handleProperty(target, key, descriptor) {
+  const superDesc = findSuperDesc(target, key);
 
   return {
     ...superDesc,
-    value: descriptor.value,
-    initializer: descriptor.initializer,
     get: descriptor.get || superDesc.get,
     set: descriptor.set || superDesc.set
   };
 }
 
+function findSuperDesc(target, key) {
+  let result;
+  if (typeof target === 'function') {
```

Comment:
could be minimized by ``result = findSuperDescFrom...(target, key, descriptor, typeof target === 'object')``